validphys2/src/validphys/closuretest/closure_checks.py

```diff
@@ -77,6 +77,14 @@ def check_t0pdfset_matches_law(t0pdfset, fit):
             f"Underlying pdf: {t0_from_fit}, does not match t0pdfset: {t0pdfset}"
         )
 
+@make_argcheck
+def check_t0pdfset_matches_multiclosure_law(multiclosure_underlyinglaw, t0set):
+    """Checks that, if a multiclosure_underlyinglaw is present, it matches the t0set
+    Checks t0set instead of t0pdfset since different mechanisms can fill t0set
+    """
+    if str(t0set) != str(multiclosure_underlyinglaw):
```

Comment:
Do we need the string cast here?

Reply:
Will always be a string?
In any case, sice it (could) be a string or a PDF which knows how to `__str__` I think it doesn't do harm (maybe it does but I didn't realised)